api/test-context/src/test_context.rs

```diff
@@ -310,13 +316,30 @@ impl TestContext {
         LocalAccount::generate(self.rng())
     }
 
-    pub fn create_user_account(&self, account: &LocalAccount) -> SignedTransaction {
-        let mut tc = self.root_account();
+    pub async fn create_account(&mut self) -> LocalAccount {
+        let mut root = self.root_account().await;
+        let account = self.gen_account();
+        let factory = self.transaction_factory();
+        let txn = root.sign_with_transaction_builder(
+            factory
+                .account_transfer(account.address(), 10_000_000)
+                .expiration_timestamp_secs(u64::MAX),
+        );
+
+        let bcs_txn = bcs::to_bytes(&txn).unwrap();
+        self.expect_status_code(202)
+            .post_bcs_txn("/transactions", bcs_txn)
+            .await;
+        self.commit_mempool_txns(1).await;
+        account
+    }
+    pub async fn create_user_account(&self, account: &LocalAccount) -> SignedTransaction {
```

Comment:
Is this still useful if create_account already takes care of submitting the tx? Can we remove this somehow?

Reply:
I think it's still useful because we test different things during submitting the txn. that being said, I think we can replace some with the new create_account(). I don't want to keep growing this PR. I can do it in the following ones.